examples/6.LGT/main.jl

```diff
@@ -0,0 +1,215 @@
+
+# In this tutorial, we are interested in the so-called quantum link model (QLM) formulation of the LGT. In this formalism, depending on the configurations of the even and odd sites, the bonds between them could be interpretted as a particle $q$, an antiparticle $\bar{q}$ or a vacuum state. More specifically, the bond between an odd and an even sites corresponds to an antiparticle if both atoms are in the ground states, otherwise it is interpretted as a vacuum state. On the other hand, the bond between an even and an odd sites corresponds to a particle if both atoms are in the ground states, otherwise it is interpreted as a vacuum. Further, the Rydberg states at the odd (even) sites are interpretted as electric fields pointing to the left (right), whereas the ground states at the odd (even) sites are electric field pointing to the right (left). The electric fields correspond to the red and blue arrows in the following figure, which summarizes the mappings described above (source: [Federica M. Surace, et al.](https://journals.aps.org/prx/pdf/10.1103/PhysRevX.10.021041)).  For more details on the LGT and its mapping to the Rydberg system, the readers are encouraged to referred to the paper cited above.
+
+# ![mapping](../../../assets/LGT_mapping.png)
+
+# ## Preparing the initial state for the LGT dynamics
+
+# The LGT dynamics starts from the "anti-string" state with all electric fields pointing to the right. This is nothing but the $Z_2$ ordered state in the language of Rydberg system, and we have seen how to prepare it in previous tutorials. Here, we are interested in a 1D lattice with 21 atoms. The neighboring atoms are separated by $5.5μm$ such that they are blockaded throughout the dynamics (the typical value of the Rabi frequency is $10\pi$ MHz throughout the dynamics, which corresponds to blockade radius $R_b\approx7.46μm$. See below).  
+
+a = 5.5;
+N = 21;
+atoms = generate_sites(ChainLattice(), N, scale=a);
+subspace = blockade_subspace(atoms, a); 
+
+# In order to prepare the anti-string state of the LGT, we use piecewise linear waveforms for both the detuning and the Rabi frequency. The waveforms will last for $3.5μs$. 
+
+total_time = 3.5; 
+Ωmax = 2π * 5;
+Δmin = -2π * 10;
+Δmax = 2π * 10;
+
+Δ1 = piecewise_linear(clocks = [0.0, 0.2, total_time], values = [Δmin, Δmin, Δmax]); 
+Ω1 = piecewise_linear(clocks = [0.0, 0.2, total_time], values = [0.0, Ωmax, Ωmax]);
+
+# The waveforms for the detuning and the Rabi frequency are shown below
+fig1, (ax1, ax2) = plt.subplots(ncols = 2, figsize = (12, 4))
+Bloqade.plot!(ax1, Ω1)
+Bloqade.plot!(ax2, Δ1)
+ax1.set_ylabel("Ω1/2π (MHz)")
+ax2.set_ylabel("Δ1/2π (MHz)")
+ax1.grid()
+ax2.grid()
+fig1
+
+
+# In order to simulate the gauge theory dynamics, we define the function `Get_Ryd_density` which takes in a given set of detuning and Rabi frequency $(Δ, Ω)$, and returns the final state and the Rydberg density. 
+
+function Get_Ryd_density(Δ, Ω; dt=1e-3)
```

Comment:
Can we follow the snake case convention? This name is also a bit confusing with the API we export (rydberg_density). A longer and more self-explainable name would be nice.